### PR TITLE
[FIRE-80] Slower fire spread

### PR DESCRIPTION
**Pull request by koszar91**
Made fire spread slower by:
1. Increasing time needed for ignition from neighbor
2. Made burning cells not receive immediately full material burning temp, instead temperature is rising asymptotically to the limit.

### CODE REVIEW

File: src/main/java/pl/edu/agh/firecell/engine/algorithm/TemperaturePropagator.java [hunk 1]
```diff
@@ -89,4 +96,7 @@ private static double tempDiff(Cell cellOne, Cell cellTwo) {
         return cellOne.temperature() - cellTwo.temperature();
     }
 
+    private static boolean isCellBurning(Cell cell) {
+        return cell.flammable() && cell.burningTime() > 0;
+    }
```
Comment (K4m1lP) [inline src/main/java/pl/edu/agh/firecell/engine/algorithm/TemperaturePropagator.java:101]: This dunction is already defined in BasicAlgorithm

File: src/main/java/pl/edu/agh/firecell/engine/algorithm/TemperaturePropagator.java [hunk 2]
```diff
@@ -78,7 +78,14 @@ public double computeConvection(State oldState, Vector3i cellIndex, double curre
     }
 
     public double updateTemperatureBasedOnFire(Cell oldCell, double currentTemperature) {
-        return oldCell.burningTime() > 0 && oldCell.flammable() ? oldCell.material().getBurningTemperature() : currentTemperature;
+        if (!isCellBurning(oldCell))
+            return currentTemperature;
+
+        double diffToMaterialBurningTemperature = oldCell.material().getBurningTemperature() - currentTemperature;
+        if (diffToMaterialBurningTemperature > 0)
+            return deltaTime * diffToMaterialBurningTemperature;
```
Comment (K4m1lP) [inline src/main/java/pl/edu/agh/firecell/engine/algorithm/TemperaturePropagator.java:86]: Should return currentTemperature + deltaTime * diffToMaterialBurningTemperature

**Review by K4m1lP — CHANGES_REQUESTED** [review]
Just one bug, looks great

File: src/main/java/pl/edu/agh/firecell/engine/algorithm/TemperaturePropagator.java (hunk 1 above)
Comment (koszar91) [reply src/main/java/pl/edu/agh/firecell/engine/algorithm/TemperaturePropagator.java:101]: And that's okay :P

Comment (K4m1lP) [reply src/main/java/pl/edu/agh/firecell/engine/algorithm/TemperaturePropagator.java:103]: Why then not to use it? I don't undertand

**Comment by K4m1lP** [comment]
It it better to have HelpfulFunctions in utils or maybe in engine? And does it have the right name?

**Review by K4m1lP — APPROVED** [review]

**Review by K4m1lP — APPROVED** [review]

**Pull request merged**